src/coreclr/scripts/superpmi-setup.py

```diff
@@ -0,0 +1,395 @@
+        exclude_directories ([string]): List of folder names to be excluded.
+        exclude_files ([string]): List of files names to be excluded.
+    """
+
+    sorted_by_size = get_files_sorted_by_size(src_directory, exclude_directories, exclude_files)
+    partitions = first_fit(sorted_by_size, max_size)
+
+    index = 0
+    for p_index in partitions:
+        file_names = [curr_file[0] for curr_file in partitions[p_index]]
+        curr_dst_path = path.join(dst_directory, str(index), "binaries")
+        copy_files(src_directory, curr_dst_path, file_names)
+        index += 1
+
+
+def set_pipeline_variable(name, value):
+    """ This method sets pipeline variable.
+
+    Args:
+        name (string): Name of the variable.
+        value (string): Value of the variable.
+    """
+    define_variable_format = "##vso[task.setvariable variable={0}]{1}"
+    print("{0} -> {1}".format(name, value)) # logging
+    print(define_variable_format.format(name, value)) # set variable
+
+
+def main(main_args):
+    """ Main entrypoint
+
+    Args:
+        main_args ([type]): Arguments to the script
+    """
+    coreclr_args = setup_args(main_args)
+    source_directory = coreclr_args.source_directory
+
+    # CorelationPayload directories
+    correlation_payload_directory = path.join(coreclr_args.source_directory, "payload")
+    superpmi_src_directory = path.join(source_directory, 'src', 'coreclr', 'scripts')
+    superpmi_dst_directory = path.join(correlation_payload_directory, "superpmi")
+    arch = coreclr_args.arch
+    helix_source_prefix = "official"
+    creator = ""
+    ci = True
+    if is_windows:
+        helix_queue = "Windows.10.Arm64" if arch == "arm64" else "Windows.10.Amd64"
+    else:
+        if arch == "arm":
+            helix_queue = "(Ubuntu.1804.Arm32)Ubuntu.1804.Armarch@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-helix-arm32v7-bfcd90a-20200121150440"
+        elif arch == "arm64":
+            helix_queue = "(Ubuntu.1804.Arm64)Ubuntu.1804.ArmArch@mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-18.04-helix-arm64v8-a45aeeb-20190620155855"
+        else:
+            helix_queue = "Ubuntu.1804.Amd64"
+
+    # create superpmi directory
+    print('Copying {} -> {}'.format(superpmi_src_directory, superpmi_dst_directory))
+    copy_directory(superpmi_src_directory, superpmi_dst_directory)
+    print('Copying {} -> {}'.format(coreclr_args.core_root_directory, superpmi_dst_directory))
+    copy_directory(coreclr_args.core_root_directory, superpmi_dst_directory)
+
+    # Clone and build jitutils
+    try:
+        with tempfile.TemporaryDirectory() as jitutils_directory:
+            run_command(
+                ["git", "clone", "--quiet", "--depth", "1", "https://github.com/dotnet/jitutils", jitutils_directory])
+            # Set dotnet path to run bootstrap
+            os.environ["PATH"] = path.join(source_directory, ".dotnet") + os.pathsep + os.environ["PATH"]
+            if is_windows:
```

Comment:
Might be better to set the bootstrap string name, then only have one copy of the `run_command` line